store/store_action.go

```diff
@@ -381,11 +385,14 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 			a.Channel = ""
 		}
 
+		if a.Action == "install" || a.Action == "refresh" {
```

Comment:
For clarification, this is the critical part of the fix.

Reply:
shouldn't we just skip the if and always set it as we do other things, anway it's omitempty?

Reply:
Indeed, that's will be fine.